helix-core/src/test/java/org/apache/helix/spectator/TestRoutingDataCache.java

```diff
@@ -24,20 +24,21 @@
 import org.apache.helix.HelixConstants;
 import org.apache.helix.PropertyType;
 import org.apache.helix.TestHelper;
-import org.apache.helix.zookeeper.datamodel.ZNRecord;
 import org.apache.helix.integration.common.ZkStandAloneCMTestBase;
+import org.apache.helix.integration.manager.MockParticipantManager;
 import org.apache.helix.manager.zk.ZkBaseDataAccessor;
 import org.apache.helix.mock.MockZkHelixDataAccessor;
 import org.apache.helix.model.CurrentState;
 import org.apache.helix.tools.ClusterVerifiers.BestPossibleExternalViewVerifier;
 import org.apache.helix.tools.ClusterVerifiers.ZkHelixClusterVerifier;
+import org.apache.helix.zookeeper.datamodel.ZNRecord;
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
 public class TestRoutingDataCache extends ZkStandAloneCMTestBase {
 
-  @Test()
-  public void testUpdateOnNotification() throws Exception {
+  @Test
+  public void testUpdateOnNotification() {
```

Comment:
Good work!

Since the original issue mentions that the instability is caused by a non-deterministic execution order of tests, should we add the `dependsOnMethods` annotation to enforce the order across different environments?

Reply:
I thought about it. But since `testCurrentStatesSelectiveUpdate` does not necessarily depend on the other two tests, adding dependsOnMethods may hide flaw of test. This test should be independent with other tests -- it should not change the global/parent states that may affect other tests -- it should restore the original states after the method. 

So making it independent should be a better idea: no matter what the running order is, the tests should not fail.